frontend/app/components/Map.vue

```diff
@@ -0,0 +1,94 @@
+<script setup lang="ts">
+import { defineProps, onMounted, onUnmounted, ref } from 'vue'
+
+interface Element {
+  x: number
+  y: number
+  content: string
+}
+
+defineProps<{
+  elements: Element[]
+}>()
+
+const mapPosition = ref(0)
+const isDragging = ref(false)
+const startX = ref(0)
+const startPosition = ref(0)
```

Comment:
Yeah, and based on variable names i exactly know when `startPosition`, `startX` and `mapPosition` is used